indexer/bag_indexer/src/main/scala/uk/ac/wellcome/platform/archive/indexer/bags/BagIndexerWorker.scala

```diff
@@ -17,19 +20,39 @@ import scala.concurrent.Future
 class BagIndexerWorker(
   val config: AlpakkaSQSWorkerConfig,
   val indexer: Indexer[StorageManifest, IndexedStorageManifest],
-  val metricsNamespace: String
+  val metricsNamespace: String,
+  storageManifestDao: StorageManifestDao
 )(
   implicit
   val actorSystem: ActorSystem,
   val sqsAsync: SqsAsyncClient,
   val monitoringClient: MetricsMonitoringClient,
-  val decoder: Decoder[StorageManifest]
+  val decoder: Decoder[KnownReplicasPayload]
 ) extends IndexerWorker[
-      StorageManifest,
+      KnownReplicasPayload,
       StorageManifest,
       IndexedStorageManifest
     ](config, indexer, metricsNamespace) {
 
-  def load(source: StorageManifest): Future[StorageManifest] =
-    Future.successful(source)
+  def load(source: KnownReplicasPayload): Future[StorageManifest] = {
+
+    val bagId = BagId(
+      space = source.storageSpace,
+      externalIdentifier = source.externalIdentifier
+    )
+
+    val bagVersion = source.version
+
+    // TODO:
+    //  - actually handle errors
+    //  - fix either/future weirdness
```

Comment:
Are you planning to fix these here, or defer them to the rearrangement of these apps?

Reply:
oh here - this is a draft PR as of now